torch/optim/adadelta.py

```diff
@@ -310,7 +362,12 @@ def _multi_tensor_adadelta(
         torch._foreach_div_(deltas, std)
         torch._foreach_mul_(deltas, device_grads)
 
-        torch._foreach_add_(device_params, deltas, alpha=-lr)
+        # If LR is a tensor, the else branch will internally call item()
+        if capturable:
+            deltas_modified = torch._foreach_mul(deltas, -lr)
+            torch._foreach_add_(device_params, deltas_modified)
```

Comment:
replace with addcmul, which would fuse this, and it'll be clearer change the condition to check if the lr is a Tensor or not. This way, the noncapturable impl would also be able to accept Tensor LR.